### ISSUE TITLE
Crysis → hang after the loading screen

### ISSUE THREAD

**Issue by NerosTie** [Joshua-Ashton/d9vk#347]
Crysis hangs after the loading screen when "Shaders Quality" is set to "normal" or "high"

### Software information
You have to use the parameter "-dx9" to be in DirectX 9 mode.

### System information
- GPU: Nvidia GT 730M
- Driver: 435-17 (beta)
- Wine version: Proton 4.11
- DXVK version: #807

### Apitrace file(s)
https://mega.nz/#!K2p0UYIT!9VJPeobH8MDNxIW8kjfPZr1xK0a5zeuWVTnDcrVruz8

### Log files
[steam-17300-d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3554183/steam-17300-d3d9.log)
[steam-17300-d9vk.log](https://github.com/Joshua-Ashton/d9vk/files/3554184/steam-17300-d9vk.log)

**Comment by K0bin** [comment]
The trace crashes for me.
This may be caused by the point sprites which D9VK doesn't implement yet.
```
11039081 @0 IDirect3DDevice9::SetRenderState(this = 0x1c2088, State = D3DRS_POINTSPRITEENABLE, Value = TRUE) = D3D_OK
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state D3DRS_POINTSPRITEENABLE
11039082 @0 IDirect3DDevice9::DrawPrimitive(this = 0x1c2088, PrimitiveType = D3DPT_POINTLIST, StartVertex = 0, PrimitiveCount = 8912) = D3D_OK
```

**Comment by Joshua-Ashton** [comment]
I wasn't able to get Crysis on Steam to work or launch at all with native or d9vk (at all).

I wasn't able to get Crysis Warhead on Steam to work or launch at all with native d3d9 or d9vk (DX10 works fine tho).

**Comment by NerosTie** [comment]
It works for me... Try with "PROTON_NO_ESYNC=1"

**Comment by Joshua-Ashton** [comment]
I'm on Windows :frog:

**Comment by NerosTie** [comment]
Haaaa!

You can try to use the 64 bits binaries: 
https://steamcommunity.com/sharedfiles/filedetails/?id=316156608

**Comment by Joshua-Ashton** [comment]
Ok that made the game work and I got a crash during the parachute sequence with d9vk.

**Comment by Joshua-Ashton** [comment]
![image](https://user-images.githubusercontent.com/21316711/64277020-3c1f0300-cf41-11e9-80bc-db3a7b70e7b0.png)

![image](https://user-images.githubusercontent.com/21316711/64277051-4f31d300-cf41-11e9-8825-8eaf5131bc7b.png)


Doesn't crash in my code. I'll double check whether it crashes on native or not...

**Comment by Joshua-Ashton** [comment]
Ok this is *technically* a game bug as it crashes whenever it reaches 4GB of memory used (despite being a x64 application)

D9VK seems to use almost double the memory compared to native so I'll need to investigate why. There may be some memory and/or resource leaking going on somewhere...

**Comment by NerosTie** [comment]
On my side, the game doesn't even reach 1Go.

**Comment by Joshua-Ashton** [comment]
Well I can't replicate that.

**Comment by NerosTie** [comment]
Fixed! :tada:  (tested with #885)
(but I will have to open a new one because there is an issue with some shadows)
